File: packages/android/src/main/java/com/kozmos/components/UserLocationMarker/UserLocationMarker.figma.kt
package com.kozmos.components.userlocationmarker

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8346")
class KozmosUserLocationMarkerConnect {
    @FigmaProperty(FigmaType.Enum, "Heading")
    val showHeading: Boolean = Figma.mapping(
        "Hidden" to false,
        "Visible" to true
    )

    // heading is a bearing in degrees from the location provider; it rotates
    // the cone rather than being a Figma property.
    @Composable
    fun ComponentExample() {
        KozmosUserLocationMarker(heading = 0f, showHeading = showHeading)
    }
}

File: packages/ios/Sources/Components/UserLocationMarker/UserLocationMarker.figma.swift
import SwiftUI
import Figma

struct KozmosUserLocationMarkerConnect: FigmaConnect {
    let component = KozmosUserLocationMarker.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8346"

    @FigmaEnum(
        "Heading",
        mapping: [
            "Hidden": false,
            "Visible": true
        ]
    )
    var showHeading: Bool = false

    // heading is a bearing in degrees from the location provider; it rotates
    // the cone rather than being a Figma property.
    var body: some View {
        KozmosUserLocationMarker(heading: 0, showHeading: self.showHeading)
    }
}
